src/go/rdctl/pkg/factoryreset/delete_data.go

```diff
@@ -30,27 +30,38 @@ import (
 
 	dockerconfig "github.com/docker/docker/cli/config"
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
-	p "github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
 	"github.com/sirupsen/logrus"
 )
 
-func addAppHomeWithoutSnapshots(appHome string) []string {
-	haveSnapshots := false
-	if snapshots, err := os.ReadDir(filepath.Join(appHome, "snapshots")); err == nil {
-		haveSnapshots = len(snapshots) > 0
+func appHomeDirectories(appPaths paths.Paths) []string {
+	// These appHome subdirectories should not be deleted unless they are empty.
+	// The logic below will actually keep all of them as long as at least one is non-empty.
+	excludeDir := map[string]bool{
+		appPaths.Snapshots:       true,
+		appPaths.ContainerdShims: true,
 	}
-	if !haveSnapshots {
-		return []string{appHome}
+	haveExclusions := false
+	for dirname := range excludeDir {
+		files, err := os.ReadDir(dirname)
+		if err == nil && len(files) > 0 {
+			haveExclusions = true
+			break
+		}
+	}
+	if !haveExclusions {
+		return []string{appPaths.AppHome}
 	}
-	appHomeMembers, err := os.ReadDir(appHome)
+	appHomeFiles, err := os.ReadDir(appPaths.AppHome)
 	if err != nil {
-		logrus.Errorf("failed to read contents of dir %s: %s", appHome, err)
-		return []string{appHome}
+		logrus.Errorf("failed to read contents of dir %s: %s", appPaths.AppHome, err)
+		return []string{}
 	}
-	pathList := make([]string, 0, len(appHomeMembers))
-	for _, entry := range appHomeMembers {
-		if filepath.Base(entry.Name()) != "snapshots" {
-			pathList = append(pathList, filepath.Join(appHome, entry.Name()))
+	pathList := make([]string, 0, len(appHomeFiles))
+	for _, file := range appHomeFiles {
+		fullname := filepath.Join(appPaths.AppHome, file.Name())
+		if !excludeDir[fullname] {
```

Comment:
This leads to… fun things if we consider case-insensitive file systems. We may need to check if [`filepath.Rel`](https://pkg.go.dev/path/filepath@go1.22.0?GOOS=windows#Rel) returns `.` instead.

Reply:
I don't understand how `filepath.Rel` would help with this.

How about making the `excludeDir` entries lowercase, and the lookup lowercase as well? While slightly hackish, this should almost always do the right thing, no?

Reply:
Yeah that sounds reasonable enough. Actually, I just realized (again… I think I saw it before) that [`os.SameFile()`](https://pkg.go.dev/os@go1.22.0#SameFile) is a thing.  But I suspect you'd think that's overkill though.

Reply:
Using `os.SameFile()` sounds good, but feels complicated, as now we have to keep stat buffers for each exclusion file, but also track if they exist or not. So it feels like the logic gets more complex for little or no gain.

But if you have a suggestion how to do this elegantly, then I think I would prefer `os.SameFile()` over the lowercase option (which I already implemented, but haven't pushed yet).

Reply:
First attempt, untested:
```go
func appHomeDirectories(appPaths paths.Paths) []string {
	excludeDirs := []string{
		appPaths.Snapshots,
		appPaths.ContainerdShims,
	}
	var excludeStats []os.FileInfo
	for _, excludeDir := range excludeDirs {
		if dir, err := os.Open(excludeDir); err == nil {
			if files, err := dir.ReadDir(1); err != nil || len(files) == 0 {
				continue
			}
			if info, err := dir.Stat(); err == nil {
				excludeStats = append(excludeStats, info)
			}
		}
	}
	if len(excludeStats) == 0 {
		return []string{appPaths.AppHome}
	}
	appHomeFiles, err := os.ReadDir(appPaths.AppHome)
	if err != nil {
		logrus.Errorf("failed to read contents of dir %s: %s", appPaths.AppHome, err)
		return []string{}
	}
	pathList := make([]string, 0, len(appHomeFiles))
	for _, file := range appHomeFiles {
		if info, err := file.Info(); err == nil {
			compareFunc := func(excludeInfo os.FileInfo) bool {
				return os.SameFile(info, excludeInfo)
			}
			if !slices.ContainsFunc(excludeStats, compareFunc) {
				fullName := filepath.Join(appPaths.AppHome, file.Name())
				pathList = append(pathList, fullName)
			}
		}
	}
	return pathList
}
```